scripts/new.lua

```diff
@@ -11,174 +11,286 @@ local options = {
+            end,
+        },
+        ["configure.ac"] = {
+            deps = {"autoconf", "automake", "libtool"},
+            install = function(configs, package)
+                return [[
+        table.insert(configs, "--enable-shared=" .. (package:config("shared") and "yes" or "no"))
+        if package:debug() then
+            table.insert(configs, "--enable-debug")
+        end
+        import("package.tools.autoconf").install(package, configs)]]
+            end,
+        },
+        ["meson.build"] = {
+            deps = {"meson", "ninja"},
+            install = function(configs, package)
+                return [[
+        table.insert(configs, "-Ddefault_library=" .. (package:config("shared") and "shared" or "static"))
+        import("package.tools.meson").install(package, configs)]]
+            end,
+        },
+        ["BUILD"] = {
+            deps = {"bazel"},
+            install = function(configs, package)
+                return 'import("package.tools.bazel").install(package, configs)'
+            end,
+        },
+        ["BUILD.bazel"] = {
+            deps = {"bazel"},
+            install = function(configs, package)
+                return 'import("package.tools.bazel").install(package, configs)'
+            end,
+        },
+    }
+
     -- detect build system
+    local build_system = nil
     if repodir then
         local files = os.files(path.join(repodir, "*")) or {}
         table.join2(files, os.files(path.join(repodir, "*", "*")))
         for _, file in ipairs(files) do
             local filename = path.filename(file)
-            if filename == "xmake.lua" then
-                has_xmake = true
-            elseif filename == "CMakeLists.txt" then
-                has_cmake = true
-            elseif filename == "configure" then
-                has_autoconf = true
-            elseif filename == "autogen.sh" or filename == "configure.ac" then
-                need_autogen = true
-                has_autoconf = true
-            elseif filename == "meson.build" then
-                has_meson = true
-            elseif filename == "BUILD" or filename == "BUILD.bazel" then
-                has_bazel = true
+            if build_systems[filename] then
+                build_system = build_systems[filename]
+                break
             end
         end
         os.rm(repodir)
     end
 
     -- add dependencies
-    if has_cmake then
-        file:print("")
-        file:print('    add_deps("cmake")')
-    elseif has_meson then
-        file:print("")
-        file:print('    add_deps("meson", "ninja")')
-    elseif need_autogen then
-        file:print("")
-        file:print('    add_deps("autoconf", "automake", "libtool")')
-    elseif has_bazel then
-        file:print("")
-        file:print('    add_deps("bazel")')
+    if build_system then
+        file:print('')
+        for _, dep in ipairs(build_system.deps) do
+            file:print('    add_deps("' .. dep .. '")')
```

Comment:
please add deps in single line. like this `add_deps("autoconf", "automake", "libtool")`

you can use `table.concat(deps, "\", ")`